Introduce error type for plutus decode

waalge opened the pull request.

waalge committed:
Introduce error type for plutus decode

waalge committed:
reorder file to be inkeeping with plutus data

waalge committed:
export types for error handling

waalge committed:
fix signature to fixed array

KtorZ commented on rust/crates/cardano-tx-builder/src/cardano/plutus_data.rs:
```diff
@@ -444,104 +427,215 @@ impl<const T: usize> From<&[u8; T]> for PlutusData<'static> {
-    type Error = anyhow::Error;
+/// A first approximation of errors
+#[derive(Debug, Error)]
+pub enum PlutusDecodeError {
```
```suggestion
pub enum PlutusDataDecodeError {
```

waalge committed:
Update rust/crates/cardano-tx-builder/src/cardano/plutus_data.rs

Co-authored-by: Matthias Benkort <5680256+KtorZ@users.noreply.github.com>

waalge committed:
pr suggestions + record type in place of tuple

waalge committed:
This is not clear headed. More thought required

The error messages are loosing their context,
and PlutusDataDecodeError still looks a bit hamfisted

waalge commented:
I lost confidence this makes sense.
